### DISCUSSION TITLE
SRT Caller Provider

### DISCUSSION THREAD

**Discussion by davemevans** [OvenMediaLabs/OvenMediaEngine#1282] [Help]
To support use cases where OME needs to be the initiator of the connection, a Provider that can do SRT Caller would be really useful.

I see in the source that at least some of the relevant methods to support SRT Caller (eg `srt_connect`) are implemented in the Socket class, but the existing SRT Provider only appears to support SRT Listener.

What would be required to enable support for a Provider that can operate in Caller mode?

https://github.com/AirenSoft/OvenMediaEngine/issues/919 appears to be the same request but was closed as stale.

**Comment by getroot** [comment]
Both SRT Push Publisher and SRT Pull Provider are on our backlog. But it's just being pushed back by higher priority tasks.
